app/src/main/java/com/douglasbruce/tasky/features/photoviewer/PhotoViewerViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.douglasbruce.tasky.features.photoviewer
+
+import android.os.Parcelable
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewmodel.compose.SavedStateHandleSaveableApi
+import androidx.lifecycle.viewmodel.compose.saveable
+import com.douglasbruce.tasky.features.photoviewer.navigation.PhotoViewerArgs
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.parcelize.Parcelize
+import javax.inject.Inject
+
+@OptIn(SavedStateHandleSaveableApi::class)
+@HiltViewModel
+class PhotoViewerViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+) : ViewModel() {
+
+    private val photoViewerArgs: PhotoViewerArgs = PhotoViewerArgs(savedStateHandle)
+
+    var state by savedStateHandle.saveable {
+        mutableStateOf(
+            PhotoViewerState(
+                url = photoViewerArgs.url,
+            )
+        )
+    }
+        private set
+}
+
+@Parcelize
+data class PhotoViewerState(
+    val url: String,
+) : Parcelable
```

Comment:
This ViewModel is not really needed in the form it works right now. The UI also has direct access to `SavedStateHandle` and therefore can also just access the URL (which never changes and therefore doesn't need to be a state). 
Once you add delete functionality, this VM will be needed, but what you have right now seems to rather be a complicated way of providing an image URI